homeassistant/components/http.py

```diff
@@ -71,7 +72,8 @@
         vol.Optional(CONF_DEVELOPMENT): cv.string,
         vol.Optional(CONF_SSL_CERTIFICATE): cv.isfile,
         vol.Optional(CONF_SSL_KEY): cv.isfile,
-        vol.Optional(CONF_CORS_ORIGINS): cv.ensure_list
+        vol.Optional(CONF_CORS_ORIGINS): cv.ensure_list,
+        vol.Optional(CONF_APPROVED_IPS): cv.ensure_list
```

Comment:
You expect a list of strings, so use

```
vol.All(cv.ensure_list, [cv.string])
```

(can you also upgrade the config validation for CORS_ORIGINS?